gapis/resolve/mesh.go

```diff
@@ -30,15 +30,10 @@ func Mesh(ctx context.Context, p *path.Mesh, r *path.ResolveConfig) (*api.Mesh,
 	if err != nil {
 		return nil, err
 	}
-	mesh, err := meshFor(ctx, obj, p, r)
-	switch {
-	case err != nil:
-		return nil, err
-	case mesh != nil:
-		return mesh, nil
-	default:
-		return nil, &service.ErrDataUnavailable{Reason: messages.ErrMeshNotAvailable()}
+	if mesh, err := meshFor(ctx, obj, p, r); err != api.ErrMeshNotAvailable {
+		return mesh, err
```

Comment:
It looks like if a non-ErrMeshNotAvailable error happen, the mesh will still get returned, is this expected?

Reply:
Yes, because then the err is either nil and the mesh is good, or there was another kind of error, which should be surfaced.